### DISCUSSION TITLE
Updating files from website

### DISCUSSION THREAD

**Discussion by Moosems** [pyinstaller/pyinstaller#7577] [Help] [answered]
If I make an app (lets say an IDE) and I want to update files (syntax builds), how could I have some pre-builts come with the app and then update them when connecting to the internet? Would you say save the most recent builds to a folder in `Application Support` and, when first booting, check if there's build there and if not load the builtin build in the Application while saving them to the `Application Support` dir?

**Comment by rokm** [comment, marked as answer]
That sounds like a reasonable approach. A variation of that would be to check the external directory in `Application Support` and if it does not exist, fall back to built-in files (i.e., without copying them over to `Application Support`).

**Reply by Moosems** [reply]
And when connected to the internet, if it finds that the built-ins are outdated then download the new versions to `Application Support`?

**Reply by rokm** [reply]
It's up to you to work out what your update strategy should be.

The bottom line in, macOS app bundles are supposed to be immutable, so you have to use external location (e.g., application-specific directory in `Application Support`) to store your extra data from internet.

**Reply by Moosems** [reply]
I ask because PyUpdater has become deprecated (though it didn't have very good documentation to begin with) and as such I need to figure out my own way for the app to update. Are there any methods you know of for updating the actual application besides saving the real application to application support and using a bootloader to update those files?

**Reply by rokm** [reply]
> I ask because PyUpdater has become deprecated (though it didn't have very good documentation to begin with) and as such I need to figure out my own way for the app to update. Are there any methods you know of for updating the actual application besides saving the real application to application support and using a bootloader to update those files?

I've never had to look into updating methods, so I have no idea (but I can easily imagine different OS mechanisms getting in the way of in-place application updates).

**Reply by Moosems** [reply]
Updating apps like this seems very ticky and I still can't figure out how other applications like VS Code do it without modifying the bundle.
